### PR TITLE
Add option to enable smart case matching.

### PR DESCRIPTION
**Pull request by kristijanhusak**
This PR enables smart case matching.

### CODE REVIEW

File: doc/completion-nvim.txt [hunk 1]
```diff
@@ -346,15 +346,23 @@ g:completion_matching_strategy_list	*g:completion_matching_strategy_list*
...
 >
 	let g:completion_matching_ignore_case = 1
 <
 	default value: 0
 
+g:completion_matching_smart_case	*g:completion_matching_smart_case*
+
+	Enable smart case matching in all matching strategy. For example
+>
+	let g:completion_matching_smart_case = 1
```
Comment (lithammer) [inline doc/completion-nvim.txt:360]: Would it make sense to change this to default to the value of `smartcase`?

```suggestion
	let g:completion_matching_smart_case = &smartcase
```

Comment (kristijanhusak) [reply doc/completion-nvim.txt:360]: I guess that would be ok, but it's up to @haorenW1025 .
We do not use `&ignorecase` for the `g:completion_matching_ignore_case` option though...

Comment (lithammer) [reply doc/completion-nvim.txt:360]: > We do not use `&ignorecase` for the `g:completion_matching_ignore_case` option though...

True. But I guess that could also be a valid change to make (possibly separate from this though)?

Comment (haorenW1025) [reply doc/completion-nvim.txt:360]: Hmm using `&smartcase` is a great idea, I can change `g:completion_matching_ignore_case` to use`&ignore` by default too.

Comment (kristijanhusak) [reply doc/completion-nvim.txt:360]: I updated it to use `&smartcase`.

**Comment by haorenW1025** [comment]
LGTM, thanks again for your contribution:) Highly appreciate!

**Comment by kristijanhusak** [comment]
I think I didn't implement this properly, at least not in the same way Vim does it. Help says this for `smartcase`:

```
'smartcase' 'scs'	boolean	(default off)
			global
	Override the 'ignorecase' option if the search pattern contains upper
	case characters.  Only used when the search pattern is typed and
	'ignorecase' option is on.
```

Since now we default these settings to `&ignorecase` and `&smartcase`, i think we should be doing the same thing built in functionality does. What are your thoughts @haorenW1025 ?

**Pull request merged**